Offseason2022/src/main/java/frc/robot/commands/Auto1Ball2OppLeft.java

```diff
@@ -3,32 +3,106 @@
+      Vision vision)
+  {
+    setName("AutoBall2OppLeft");
+
+    // @formatter:off
+    addCommands(
+      new PrintCommand("AUTO 1 BALL 2 OPP LEFT - START"),
+      // wait timer set in SmarDashboard new PrintCommand("WAIT"), new AutoWait(1), // Deploy Intake
+      new PrintCommand("Deply intake"), 
+      new ParallelDeadlineGroup(
+       new IntakeDeploy(intake, true), 
+       new AutoStop(drivetrain)
+       ), 
+      // Drive to a shooting position 
+      new PrintCommand("drive to a shooting position"), 
+      new ParallelDeadlineGroup(
+        new WaitUntilCommand(drivetrain::driveWithPathFollowerIsFinished
+      ), 
+      new AutoDrivePath(drivetrain, AUTOConstants.k1Ball2OppLeft_path2, true), 
+      new ScoringPrime(shooter, vision)
+      ), 
+      // Shoot preloaded ball 
+      new PrintCommand("Shoot preloaded ball"), 
+      new ParallelDeadlineGroup(
+        new ScoringActionUpperHub(intake, fConv, tConv, shooter, 1.0), 
+        new AutoStop(drivetrain)
+      ), 
+      // Drive to 1st opponent's ball and intake 
+      new PrintCommand("Drive to 1st opponent's ball and intake"), 
+      new ParallelDeadlineGroup(
+        new WaitUntilCommand(drivetrain::driveWithPathFollowerIsFinished), 
+        new AutoDrivePath(drivetrain, AUTOConstants.k1Ball2OppLeft_path2, false), 
+        new IntakingAction(intake, fConv, tConv)
+      ), 
+      // Second wait timer set in SmartDashboard 
+      new PrintCommand("WAIT"), 
+      new AutoWait(AutoTimer.TIMER2), 
+      // Drive to 2cnd opponent's ball and intake 
+      new PrintCommand("Drive to 2cnd opponent's ball and intake"), 
+      new ParallelDeadlineGroup(
+        new WaitUntilCommand(drivetrain::driveWithPathFollowerIsFinished), 
+        new AutoDrivePath(drivetrain, AUTOConstants.k1Ball2OppLeft_path3, false), 
+        new ScoringPrime(shooter, vision), 
+        new IntakingAction(intake, fConv, tConv)
+      ), 
+      // stow intake new PrintCommand("Stow intake"), 
+      new ParallelDeadlineGroup(
+        new IntakeDeploy(intake, false), 
+        new AutoStop(drivetrain)
+      ),
+      //PUT IN CONDITIONAL COMMAND: LITERALLY IF THIS RETURN THIS OR LIKE RETURN THE BOOLEAN 
+      // Shoot opponent's ball
+      new ConditionalCommand(
+        new ParallelDeadlineGroup(
+          new ScoringActionLowerHub(intake, fConv, tConv, shooter, 2), 
+          new AutoStop(drivetrain), 
+          new PrintCommand("Shoot opponent's ball")
+        ), 
+      new AutoStop(drivetrain),
+      this::dummy
+      ), 
+       // Stop shooting and driving 
+      new PrintCommand("Stop shooting"), 
+      new ParallelDeadlineGroup(
+        new ScoringStop(intake, fConv, tConv, shooter, vision), 
+        new PrintCommand("AUTO 3 BALL LEFT - END")
+      )
     );
+      // @formatter:on
   }
 
   @Override
-  public boolean runsWhenDisabled( )
+  public final boolean runsWhenDisabled( )

```

Comment:
Don't think this should be final, but we need to get this merged ASAP. Formatting and print commands look pretty good.